### DISCUSSION TITLE
Find a Specific Edge Between Two Vertices

### DISCUSSION THREAD

**Discussion by ahbagheri** [Gremlinq/ExRam.Gremlinq#646] [General]
Given two nodes with `id1` and `id2`, how can I find a Specific Edge of Type `E` between those two?

**Comment by bernardodiasr** [comment]
Hi,

You need to ensure which is the connection direction.
We will suppose:  id1->id2

I think you can use:  

g.V('id1').OutE('SpecificEdge') : returns all the specific edge(s) that are coming out of the vertex itself (id1)
g.V('id1').OutE() : returns all the edges that are coming out of the vertex itself (id1)

you can do the same but with id2 vertex.

g.V('id2').InE('SpecificEdge') 
g.V('id2').InE() 

With ExRam.Gremlinq.Core:

**var edge = await _g.V("id1").OutE<SpecificEdge>();**

or

**var edge = await _g.V("id2").InE<SpecificEdge>();**

Hope this helps!

**Reply by bernardodiasr** [reply]
You can use also:

**g.V('id1').outE('SpecificEdge').as('e').inV().has('id','id2').select('e')**

**Reply by danielcweber** [reply]
That'll work. You could also use a filter:

g.V('id').outE('SpecificEdge').where(inV().has('id', 'id2'))
